File: Clase 16/JAVA/ejemploAnimales/Animal.java
package ejemploAnimales;

public class Animal{
    protected int id;
    protected String nombre;
    protected String especie;

    public Animal(int id, String nombre, String especie){
        this.id = id;
        this.nombre = nombre;
        this.especie = especie;
    }

    public void seMueve(){
        System.out.println(this.nombre + " se esta moviendo");
    }

    public void haceSonido(){
        System.out.print(this.nombre + " hace un sonido...");
    }
}

File: Clase 18/PYTHON/ejemplo_animales/Animal.py
class Animal:

    # Polimorfismo parametrico
    # parametros opcionales
    def __init__(self, id:int, nombre=None, especie=None):
        self.id = id
        self.nombre = nombre if nombre is not None else ""
        self.especie = especie if especie is not None else ""
    

    def se_mueve(self):
        if self.nombre == "":
            print(f"Animal X se mueve...")
        else:
            print(f"{self.nombre} se mueve...")

    def hace_sonido(self):
        if self.nombre == "":
            print(f"Animal X hace un sonido...")
        else:
            print(f"{self.nombre} hace un sonido...")